tests/Examples/plaintext/dot_product/dot_product_8_mod_test.c

```diff
@@ -0,0 +1,30 @@
+#include <stdint.h>
+#include <stdio.h>
+
+// This is the function we want to call from LLVM
+// For mod-arith, the width is 64 bits
+int64_t dot_product(
+    /* arg 0*/
+    int64_t *allocated, int64_t *aligned, int64_t offset, int64_t size,
+    int64_t stride,
+    /* arg 1*/
+    int64_t *allocated_2, int64_t *aligned_2, int64_t offset_2, int64_t size_2,
+    int64_t stride_2);
+
+int main() {
+  int64_t arg0[8] = {1, 2, 3, 4, 5, 6, 7, 8};
+  int64_t arg1[8] = {2, 3, 4, 5, 6, 7, 8, 9};
+  int64_t expected = 240;
```

Comment:
I think it would be worth choosing a plaintext modulus that is small enough in one of the tests so that the expected value is different from the test in which no modulus is set. How about a modulus of `17`, so that the expected value is `2`?

Another thing to make this easier is to have this `main` print out `res`, and then have the test assertion happen in FileCheck using `--check-prefix=CHECK_MY_UNIQUE_PREFIX` as you do in other lit tests in this file.

[edit]: I noticed you did this for roberts cross, but having it for a smaller test might be better so regressions could be caught by smaller, faster running tests.

Reply:
Added the modulus=17 test case